src/main/java/io/vertx/ext/shell/cli/impl/CliTokenImpl.java

```diff
@@ -0,0 +1,122 @@
+package io.vertx.ext.shell.cli.impl;
+
+import io.termd.core.readline.LineStatus;
+import io.vertx.ext.shell.cli.CliToken;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class CliTokenImpl implements CliToken {
+
+  final boolean text;
+  final String raw;
+  final String value;
+
+  public CliTokenImpl(boolean text, String value) {
+    this(text, value, value);
+  }
+
+  public CliTokenImpl(boolean text, String raw, String value) {
+    this.text = text;
+    this.raw = raw;
+    this.value = value;
+  }
+
+  @Override
+  public boolean isText() {
+    return text;
+  }
+
+  @Override
+  public boolean isBlank() {
+    return !text;
+  }
+
+  public String raw() {
+    return raw;
+  }
+
+  public String value() {
+    return value;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
equals is used because it's useful in unit tests, not really because it's put in a map